tests/plugins/link/anchor.js

```diff
@@ -207,6 +208,56 @@
 
 				assert.beautified.html( expected, editor.getData(), 'Prevent duplicated anchors failed in the ordered list with styled word' );
 			} );
+		},
+		// (#4802)
+		'test prevent adding anchor with SPACE character': function() {
+			assertWhitespaceAnchor( this.editorBot, '\u0020', 'SPACE' );
+		},
+
+			// (#4802)
+		'test prevent adding anchor with CHARACTER TABULATION character': function() {
+			assertWhitespaceAnchor( this.editorBot, '\u0009', 'CHARACTER TABULATION' );
+		},
+
+			// (#4802)
+		'test prevent adding anchor with FORM FEED character': function() {
+			assertWhitespaceAnchor( this.editorBot, '\u000c', 'FORM FEED' );
+		},
+
+			// (#4802)
+		'test add anchor with non-breaking space': function() {
+			var bot = this.editorBot,
+				windowStub = sinon.stub( window, 'alert' ),
+				template = '[<p>Simple text</p>]';
+
+			bot.setHtmlWithSelection( template );
+			bot.dialog( 'anchor', function( dialog ) {
+				dialog.setValueOf( 'info', 'txtName', 'Foo\u00a0bar' );
+				dialog.getButton( 'ok' ).click();
+
+				assert.areEqual( 0, windowStub.callCount );
+			} );
+
+			windowStub.restore();
```

Comment:
Probably moving the restoration to the `tearDown()` function would be more convenient.

Otherwise, it would be good to put the restoration before any assertions. Currently, if assertions fail, the `alert()` stub won't be restored, leading to errors when stubbing the `alert()` in subsequent tests.

Reply:
Moved restoration before assertion because IMO we don't need to tear down before any tests.